app/models/product_import/entry_processor.rb

```diff
@@ -227,8 +233,8 @@ def permission_by_id?(supplier_id)
       @editable_enterprises.value?(Integer(supplier_id))
     end
 
-    def import_into_inventory_by_supplier?(supplier_id)
-      @import_settings[:settings] && @import_settings[:settings][supplier_id.to_s] && @import_settings[:settings][supplier_id.to_s]['import_into'] == 'inventories'
+    def importing_into_inventory?
+      @import_settings[:settings] && @import_settings[:settings]['import_into'] == 'inventories'
```

Comment:
not for this PR, but I think it's time to move the hash keys `settings, enterprise_to_reset and updated_ids` as ivars of this class specified in the constructor. Your fingers must hurt after typing all these hashes :trollface: 

As a result, `EntryProcessor` won't depend on the data structure of `@import_settings` anymore and that is a big win. If we happen to change it in the future (chances are that we will), it won't break this class.

Reply:
Now I see this class already has tons of ivars. That is a code :nose: . Before doing what I suggested, we should understand if some of these vars are hiding one or more concepts that deserve their own object.